### PR TITLE
Scan: Implemented the data msg cat to tx index page

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #2679

## Implementation details

- Implemented these msg boxes to Tx index page (both success and fail msg)
  - CreateDataSourceMsg
  - EditDataSourceMsg
  - CreateOracleScriptMsg
  - EditOracleScriptMsg 
  - RequestMsg
  - ReportMsg
- Refactor the key-value table to be more simple and easier to use
- Updated `.gitignore`

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/components/KVTable.re [hunk 1]
```diff
@@ -166,90 +78,39 @@ let renderField = (field, maxWidth, isRight) => {
...
-             </Col>
+let make = (~headers=["Key", "Value"], ~rows) => {
+  let columnSize = headers |> Belt_List.length > 2 ? Col.Four : Col.Six;
+  let valueWidth = Media.isMobile() ? 70 : 480;
+  <div className=Styles.tabletContainer>
+    <div className=Styles.tableSpacing>
+      <Row.Grid>
+        {headers
+         ->Belt_List.mapWithIndex((i, header) => {
+             <Col.Grid key={i |> string_of_int} col=columnSize colSm=columnSize>
```
Comment (evilpeach) [inline scan/src/components/KVTable.re]: Let's use `header+i` as key

File: scan/src/components/KVTable.re [hunk 2]
```diff
@@ -166,90 +78,39 @@ let renderField = (field, maxWidth, isRight) => {
...
-                       isRight,
-                     )}
-                  </div>
-                </Col>
-              })
-            ->Belt_List.toArray
-            ->React.array}
+    {rows
+     ->Belt.List.mapWithIndex((i, row) => {
+         <div key={i |> string_of_int} className={Css.merge([Styles.tableSpacing])}>
```
Comment (evilpeach) [inline scan/src/components/KVTable.re]: Let's use something more than index.

File: scan/src/components/KVTable.re [hunk 3]
```diff
@@ -166,90 +78,39 @@ let renderField = (field, maxWidth, isRight) => {
...
+           <Row.Grid>
+             {row
+              ->Belt_List.mapWithIndex((j, value) => {
+                  <Col.Grid key={j |> string_of_int} col=columnSize colSm=columnSize>
```
Comment (evilpeach) [inline scan/src/components/KVTable.re]: Let's use something more than index.

File: scan/src/components/tx-index/IndexDataMsg.re [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        <Heading value="Name" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}>
+          <TypeID.DataSource id={dataSource.id} />
+          <HSpacing size=Spacing.sm />
+          <Text value={dataSource.name} />
+        </div>
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
```
Comment (evilpeach) [inline scan/src/components/tx-index/IndexDataMsg.re]: Do we handle `[do-not-modify]` case?

File: scan/src/components/tx-index/IndexDataMsg.re [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        <Heading value="Name" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}>
+          <TypeID.OracleScript id={oracleScript.id} />
+          <HSpacing size=Spacing.sm />
+          <Text value={oracleScript.name} />
+        </div>
+      </Col.Grid>
+    </Row.Grid>;
+  };
+};
```
Comment (evilpeach) [inline scan/src/components/tx-index/IndexDataMsg.re]: Do we handle `[do-not-modify]` case?

File: scan/src/components/tx-index/IndexDataMsg.re [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+module RequestMsg = {
+  [@react.component]
+  let make = (~request: TxSub.Msg.Request.success_t) => {
+    let calldataKVsOpt = Obi.decode(request.schema, "input", request.calldata);
```
Comment (evilpeach) [inline scan/src/components/tx-index/IndexDataMsg.re:118]: As I found out the design missed the request ID. Please add it.

File: scan/src/components/tx-index/IndexDataMsg.re [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        <Heading value="Owner" size=Heading.H5 marginBottom=8 />
+        <AddressRender address={report.reporter} />
+      </Col.Grid>
+      <Col.Grid col=Col.Six mb=24>
+        <Heading value="Request ID" size=Heading.H5 marginBottom=8 />
+        <div className={CssHelper.flexBox()}> <TypeID.Request id={report.requestID} /> </div>
+      </Col.Grid>
+      <Col.Grid>
+        <Heading value="Raw Data Report" size=Heading.H5 marginBottom=8 />
+        <KVTable
```
Comment (evilpeach) [inline scan/src/components/tx-index/IndexDataMsg.re]: Can we add request id as well?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/tx-index/IndexDataMsg.re (hunk 7 above)
Comment (pzshine) [reply scan/src/components/tx-index/IndexDataMsg.re]: There still has the requestID on line 218, but I've removed the container.

**Review by evilpeach — APPROVED** [review]

**Pull request merged**